### PR TITLE
Serve docs using docker to remove mac dependency

### PR DESCRIPTION
**Pull request by sudo-bmitch**
First step for me updating the docs is having a way to run them locally. This switches to docker so we don't have the MacOS dependency.

Signed-off-by: Brandon Mitchell <git@bmitch.net>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -110,13 +110,9 @@ example-sample-pipeline: ## Run the sample-pipeline example
 example-ibm-tutorial: ## Run the IBM pipeline example
 	bash examples/ibm-tutorial/ibm-tutorial.sh
 
-.PHONY: docs-setup
-docs-setup: ## Install the tool to build the documentation
-	bash docs/bootstrap.sh
```
Comment (rgreinho) [inline Makefile:115]: I think we can remove this script then.

**Review by rgreinho — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (sudo-bmitch) [reply Makefile:115]: Doh, I had done that but VS code was sitting at a commit message editor. Gonna take me a bit to get used to these new prompts.

**Review by rgreinho — APPROVED** [review]

**Comment by bradbeck** [comment]
I get the following error when running `make doc-serve`
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Os { code: 9, kind: Uncategorized, message: "Bad file descriptor" })', src/cmd/serve.rs:327:59
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [docs-serve] Error 101
```

File: docs/serve.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# Define variables.
+C_GREEN='\033[32m'
+C_RESET_ALL='\033[0m'
+
+GIT_ROOT=$(git rev-parse --show-toplevel)
+: "${ZOLA_PORT:=8080}"
+
+echo -e "${C_GREEN}Serving documentation on http://localhost:${ZOLA_PORT}/ (use cont-c to stop)...${C_RESET_ALL}"
+docker run \
```
Comment (bradbeck) [inline docs/serve.sh:12]: Maybe add `--rm` to cleanup the containers when they stop?

**Review by bradbeck — APPROVED** [review]

**Review by rgreinho — APPROVED** [review]

**Pull request merged**